src/sql_engine/src/dml/select.rs

```diff
@@ -45,107 +45,115 @@ impl<'sc> SelectCommand<'sc> {
 
-                if let Some(column_definition) = found {
-                    column_definitions.push(column_definition);
-                } else {
-                    non_existing_columns.push(column_name.clone());
+                if !non_existing_columns.is_empty() {
+                    self.session
+                        .send(Err(QueryError::column_does_not_exist(non_existing_columns)))
+                        .expect("To Send Result to Client");
+                    return Ok(());
                 }
-            }
 
-            if !non_existing_columns.is_empty() {
+                let values: Vec<Vec<String>> = records
+                    .map(Result::unwrap)
+                    .map(Result::unwrap)
+                    .map(|(_key, values)| {
+                        let row: Vec<String> = values.unpack().into_iter().map(|datum| datum.to_string()).collect();
+
+                        let mut values: Vec<(&usize, String)> = vec![];
+                        for (_i, (origin, ord)) in column_indexes.iter().enumerate() {
+                            for (index, value) in row.iter().enumerate() {
+                                if index == *origin {
+                                    values.push((ord, value.clone()))
+                                }
+                            }
+                        }
+                        log::debug!("{:#?}", values);
+                        values.into_iter().map(|(_, value)| value).collect()
+                    })
+                    .collect();
+
+                let projection = (
+                    description
+                        .into_iter()
+                        .map(|column_definition| (column_definition.name(), (&column_definition.sql_type()).into()))
+                        .collect(),
+                    values,
+                );
                 self.session
-                    .send(Err(QueryError::column_does_not_exist(non_existing_columns)))
-                    .expect("To Send Result to Client");
-                return Err(SystemError::runtime_check_failure("Column Does Not Exist".to_owned()));
+                    .send(Ok(QueryEvent::RecordsSelected(projection)))
+                    .expect("To Send Query Result to Client");
+                Ok(())
             }
-
-            let description = column_definitions
-                .into_iter()
-                .map(|column_definition| (column_definition.name(), (&column_definition.sql_type()).into()))
-                .collect();
-
-            Ok(description)
-        } else {
-            self.session
-                .send(Err(QueryError::feature_not_supported(self.raw_sql_query.to_owned())))
-                .expect("To Send Query Result to Client");
-            Err(SystemError::runtime_check_failure("Feature Not Supported".to_owned()))
         }
     }
 
-    pub(crate) fn execute(&mut self) -> SystemResult<()> {
+    fn parse_select_input(&self) -> SystemResult<SelectInput> {
```

Comment:
@AndrewBregger am I right that we'll have to use/move this function to `QueryProcssor`?
This is cool that we discover one more case to think for query planning 👍